src/lib.rs

```diff
@@ -645,6 +682,15 @@ impl SystemHandle {
         }
     }
 
+    pub fn publish<E: 'static + std::any::Any>(&self, event: E) {
+        let event_subscribers = self.event_subscribers.lock();
+        if let Some(subs) = event_subscribers.events.get(&TypeId::of::<E>()) {
+            for sub in subs {
+                sub(&event);
```

Comment:
The one thing I don't like about this is that it could get very costly (imagine a big list of subscribers with a big event type being cloned and converted into their respective message types), and this cost is paid upfront by the caller to `publish`. 

It feels like for the general use case of a pub/sub framework, a high priority thread that's doing the publishing won't be able to bear this cost, so it seems like there's an abstraction missing in the middle (some sort of queue that takes these published events and distributes them to the subscribers). This middleman then could be prioritized appropriately so it doesn't slow down the high priority thread that just needs to publish and keep going.

Reply:
After local conversation:

It may be interesting to, instead of sending the whole topic struct, instead send a reference counted handle to the same heap data to all of the subscribers, which is a lot less costly and would alleviate most of my issues with this (though I'm still a bit worried about deadlocks).

Reply:
@strohel and I discussed this a bit, and our conclusion was that the decision is probably better left to the users of the API, so they can wrap their message in an `Arc`, or not if they don't mind the clones.

Reply:
I've also realized that forcing `Arc` would be limiting: it requires `T` to be `Sync` for `Arc<T>` to be `Send`, whereas currently we only require `Event` to be `Send` and not necessarily `Sync`.